selectors/selectors.md

```diff
@@ -108,3 +108,269 @@ These components imply or expand into the following things:
 - Allow language-specific implementations of selectors (parsers, execution, etc).
 - (IMPORTANT) Well-designed set of test vectors representing a variety of use cases for IPLD Selectors.
 - A recommended structure for implementing a selector type, with an easy to use test suite.
+
+
+Selector types
+--------------
+
+### CID Selector
+
+Selects a single CID.
+
+
+### Path Selector
+
+Selects a single path segment.
+
+
+### Array Selector
+
+Selects a slice or single item of an array of items.
+
+
+### Recursive Selector
+
+Uses another Selector to select things recursively. It also takes a limit on how often this recursion should happen (the limit can be set to "infinity" to keep traversing until there are no matching paths left.
+
+The recursion may also be stopped by a given CID.
+
+
+### CID Rooted Selector
```

Comment:
I introduced the CID Rooted Selector so that we can have other ones, like a sequential one (i'd call it multi selector) in the future. It can be added to the spec later on. I just wanted to keep things as minimal as possible to make agreement easier.

Reply:
You can use `{"root": CID, selector: []}` to select only a CID.

Reply:
About the sequential selector: https://github.com/ipld/specs/issues/107

Reply:
My point was to keep each selector as minimal as possible. The CID rooted selector does two things: selects a CID and a sequence of follow-on selectors.

Reply:
Yes. Currently it's the only "entry point". So the CID Rooted Selector is kind of the only proper Selector, all other ones are "Selector Components" or something like that.